camel/memories/blocks/chat_history_block.py

```diff
@@ -11,68 +11,66 @@
 
-    This memory class helps manage conversation histories with a designated
-    storage mechanism, either provided by the user or using a default
+    This memory block helps manage conversation histories with a key-value
+    storage backend, either provided by the user or using a default
     in-memory storage. It offers a windowed approach to retrieving chat
     histories, allowing users to specify how many recent messages they'd
     like to fetch.
 
-    `ChatHistoryMemory` requires messages to be stored with certain
+    :class:`ChatHistoryBlock` requires messages to be stored with certain
     metadata (e.g., `role_at_backend`) to maintain consistency and validate
     the chat history.
 
     Args:
-        context_creator (BaseContextCreator): A context creator contianing
-            the context limit and the message pruning strategy.
         storage (BaseKeyValueStorage, optional): A storage mechanism for
             storing chat history. If `None`, an :obj:`InMemoryKeyValueStorage`
             will be used. (default: :obj:`None`)
-        window_size (int, optional): Specifies the number of recent chat
-            messages to retrieve. If not provided, the entire chat history
-            will be retrieved. (default: :obj:`None`)
     """
 
     def __init__(
         self,
-        context_creator: BaseContextCreator,
         storage: Optional[BaseKeyValueStorage] = None,
-        window_size: Optional[int] = None,
     ) -> None:
-        self.context_creator = context_creator
         self.storage = storage or InMemoryKeyValueStorage()
-        self.window_size = window_size
 
-    def get_context(self) -> Tuple[List[OpenAIMessage], int]:
-        r"""Gets chat context with a proper size for the agent from the memory
+    def retrieve(
+        self,
+        window_size: Optional[int] = None,
+    ) -> List[ContextRecord]:
+        r"""Retrieves records with a proper size for the agent from the memory
         based on the window size or fetches the entire chat history if no
         window size is specified.
 
+        Args:
+            window_size (int, optional): Specifies the number of recent chat
+                messages to retrieve. If not provided, the entire chat history
+                will be retrieved. (default: :obj:`None`)
+
         Returns:
-            (List[OpenAIMessage], int): A tuple containing the constructed
-                context in OpenAIMessage format and the total token count.
+            List[ContextRecord]: A list of retrieved records.
+
         Raises:
-            ValueError: If the memory is empty or if the first message in the
-                memory is not a system message.
+            ValueError: If the memory is empty.
         """
         record_dicts = self.storage.load()
         if len(record_dicts) == 0:
-            raise ValueError("The `ChatHistoryMemory` is empty.")
+            warnings.warn("The `ChatHistoryMemory` is empty.")
```

Comment:
from the docstring it will raise ValueError, but the implementation instead uses warnings.warn and returns an empty list